src/vue/forms/MassPaymentForm.vue

```diff
@@ -0,0 +1,266 @@
+              :error-message="getFieldErrorMessage('form.assetCode')"
+              :disabled="formMixin.isDisabled"
+            >
+              <option
+                v-for="asset in transferableBalancesAssets"
+                :key="asset.code"
+                :value="asset.code"
+              >
+                {{ asset.name }}
+              </option>
+            </select-field>
+
+            <template v-if="form.assetCode">
+              <p class="app__form-field-description">
+                {{
+                  'mass-payment-form.available-for-payment-hint' | globalize({
+                    amount: {
+                      value: accountBalance.balance,
+                      currency: form.assetCode
+                    }
+                  })
+                }}
+              </p>
+            </template>
+          </div>
+        </div>
+
+        <div class="app__form-row">
+          <div class="app__form-field">
+            <amount-input-field
+              v-model="form.amount"
+              name="mass-payment-amount"
+              validation-type="outgoing"
+              :label="'mass-payment-form.amount-lbl' | globalize"
+              :asset="assetByCode(form.assetCode)"
+              :disabled="formMixin.isDisabled"
+            />
+          </div>
+        </div>
+
+        <div class="app__form-actions">
+          <button
+            v-ripple
+            :disabled="formMixin.isDisabled"
+            class="app__form-submit-btn app__button-raised"
+            type="submit"
+          >
+            {{ 'mass-payment-form.submit-btn' | globalize }}
+          </button>
+        </div>
+      </form>
+    </template>
+
+    <template v-else>
+      <no-data-message
+        class="mass-payment-form__no-data-message"
+        :title="'mass-payment-form.no-assets-msg-title' | globalize"
+        :message="'mass-payment-form.no-assets-msg-description' | globalize"
+      />
+    </template>
+  </div>
+</template>
+
+<script>
+import FormMixin from '@/vue/mixins/form.mixin'
+import IdentityGetterMixin from '@/vue/mixins/identity-getter'
+import NoDataMessage from '@/vue/common/NoDataMessage'
+import { required } from '@validators'
+
+import { CsvUtil } from '@/js/utils/csv.util'
+import { ErrorHandler } from '@/js/helpers/error-handler'
+import { Bus } from '@/js/helpers/event-bus'
+
+import { mapGetters, mapActions } from 'vuex'
+import { store, vuexTypes } from '@/vuex'
+
+import { api } from '@/api'
+import { base } from '@tokend/js-sdk'
+import { MathUtil } from '@/js/utils'
+
+const EVENTS = {
+  submitted: 'submitted',
+}
+
+export default {
+  name: 'mass-payment-form',
+
+  components: { NoDataMessage },
+
+  mixins: [FormMixin, IdentityGetterMixin],
+
+  props: {
+    assetCode: { type: String, default: '' },
+    receivers: { type: String, default: '' },
+    amount: { type: [String, Number], default: '' },
+  },
+
+  data () {
+    return {
+      form: {
+        assetCode: this.assetCode ||
```

Comment:
wtf?!?! use mapGetters for `vuexTypes.transferableBalancesAssets`, and methods for calculate `form.assetCode`

Reply:
used